specification.markdown

```diff
@@ -245,6 +245,18 @@ A `Binary Element` MUST declare a length in octets from zero to `VINTMAX`.
 
 The contents of a `Binary Element` should not be interpreted by the `EBML Reader`.
 
+# Terminating Elements
+
+`Null Octets`, which are octets with all bits set to zero, MAY follow the value of a `String Element` or `UTF-8 Element` to serve as a terminator. An `EBML Writer` MAY terminate a `String Element` or `UTF-8 Element` with `Null Octets` in order to overwrite a stored value with a new value of lesser length while maintaining the same `Element Data Size` (this can prevent the need to rewrite large portions of an `EBML Document`); otherwise the use of `Null Octets` within a `String Element` or `UTF-8 Element` is NOT RECOMMENDED. An `EBML Reader` MUST consider the value of the `String Element` or `UTF-8 Element` to be terminated upon the first read `Null Octet` and ignore any following data within that `Element`. If an `EBML Writer` terminates the value of a `String Element` or `UTF-8 Element` with a `Null Octet` then any subsequent octets of the `Element` MUST also be `Null Octets`. A string value and a copy of that string value terminated by one or more `Null Octets` are semantically equal.
```

Comment:
What mkvpropedit and the GUI's header/chapter editors do in a case of having to shrink by one byte is to move the following element's header (ID + size field) up one byte and to extend its size field by one byte. This works because both programs only rewrite level 1 elements that way; therefore they don't have to care about having to adjust the upper level, too.

The problem with writing a null value at the end is that we've considered that to be invalid if I remember correctly as that byte wouldn't be a valid start of an element ID.

Re-writing the element to shrink and extending its size field by one byte seems to the most reasonable regarding a general description.

I wouldn't object to allowing null bytes between elements and requiring readers to simply skip such null bytes.